open-budget/app/scripts/entity_details.coffee

```diff
@@ -1,17 +1,81 @@
 class EntityDetailsView extends Backbone.View
+        events:
+          'click .exemption-expander': 'toggleExemptionDetails'
 
-        initialize: ->
-            @model.on 'ready-entity', => @render()
+        initialize:  ->
+            @model.selectedEntity.on 'change:selected', =>
+                eid = @model.selectedEntity.get('selected')
+                @entity = new Entity(pageModel: window.pageModel, entityId: eid)
+                @entity.doFetch()
+                @entity.on 'ready', => @render()
             console.log 'EntityDetailsView init'
 
         render: ->
             @$el.css('display','inherit')
-            data = @model.entity.toJSON()
+            data = @entity.toJSON()
             console.log 'render', data
             @$el.html window.JST.entity_details( data )
 
+            # for each exemption by publisher, build a view and render it, and append it
+            # to the table body
+            exemptionByPublisherBody = @$el.find('#exemption-table tbody')
+            for exemptionByPublisher in _.values(@entity.exemptionsByPublisher())
+              rowView = new ExemptionByPublisherRowView(model: exemptionByPublisher)
+              exemptionByPublisherBody.append(rowView.render().el)
+            @
+
+class ExemptionByPublisherRowView extends Backbone.View
```

Comment:
Well, ExemptionByPublisherRowView shouldn't be in the entity_details script... perhaps move it to a different file: new_spendings.coffee 

As for structure - yes, I'll do some re-org soon, it's needed indeed.